### PR TITLE
Update __init__.py

### PR DESCRIPTION
**Pull request by ahnitz**
This would modify init_logging in pycbc to work with a 'count' argument. This means if you give --verbose, the logging level is 'INFO', iv you give it twice (or -vv), it will use 'DEBUG'. 

If you keep giving (verbose) the logging level will decrease in increments in 10, but those don't have canonical names, and would be up to the user to use at their volition.

### CODE REVIEW

File: pycbc/__init__.py
Comment (tdent) [inline pycbc/__init__.py:98]: Seems OK, except that the code is now a bit cryptic. Suggest to add a comment making people consult https://docs.python.org/3/library/logging.html#levels to explain what is happening. 
Also, `logging.WARN` seems to be deprecated in favor of `logging.WARNING` so can we have that.

Comment (tdent) [inline pycbc/__init__.py:98]:
```suggestion
    initial_level = logging.WARNING
```

File: pycbc/__init__.py
Comment (tdent) [inline pycbc/__init__.py:100]: Actually I think it could all be done in a single line:
`initial_level = logging.WARNING - 10 * int(verbose)`
since False maps to 30 - 0 = WARNING, True or 1 maps to INFO and 2 maps to DEBUG

File: pycbc/__init__.py
Comment (tdent) [inline pycbc/__init__.py:100]:
```suggestion

    # See https://docs.python.org/3/library/logging.html#levels for why this works
    initial_level = logging.WARNING - int(verbose) * 10
```

**Review by tdent — APPROVED** [review]

**Comment by tdent** [comment]
FWIW, we might want to check if this works with a specific executable before merging - I will see if we can do that with the SNR optimizer fix that Praveen is working on

File: pycbc/__init__.py
Comment (GarethCabournDavies) [inline pycbc/__init__.py:99]: I think this doesn't take into account if `--verbose` uses `action='store_true'` and ends up as `False`, which worked before.

I suggest to keep the `if not verbose: elif`, or find every instance which uses `init_logging` and verbose as `store_true`.

The second is probably better but time consuming, the first would be easier to implement.

I would like to standardise logging and versioning throughout the code, which would involve the first option as it has become quite inconsistent, but that is a medium-term objective

Comment (GarethCabournDavies) [reply pycbc/__init__.py:99]: Wait - ignore that, `int(False)` is zero

Comment (tdent) [reply pycbc/__init__.py:99]: ie one of the laundry list items in https://github.com/gwastro/pycbc/issues/4483 .. 
Resolvable, I guess

**Comment by PRAVEEN-mnl** [comment]
We found if we don't supply --verbose, we get error of Nonetype object. So we made a change in the code to fix it.

**Review by tdent — CHANGES_REQUESTED** [review]
Removing my approval as this has a bug for the non-verbose case ..

File: pycbc/__init__.py
Comment (tdent) [inline pycbc/__init__.py:102]:
```suggestion
    logger = logging.getLogger()
    verbose_int = 0 if verbose is None else int(verbose)
    logger.setLevel(logging.WARNING - verbose_int * 10)  # Initial setting
```

**Review by tdent — APPROVED** [review]
Approve (assuming tests pass)

**Pull request merged**
